src/fmu/dataio/dataio.py

```diff
@@ -131,15 +131,16 @@ def _check_content(proposed: Union[str, dict]) -> Any:
     logger.info("Evaluate content")
 
     content = proposed
+    content_specific = None
     logger.debug("content is %s of type %s", str(content), type(content))
     usecontent = "unset"
     if content is None:
         warn(
-            "The <content> is not provided which defaults to 'depth'. "
+            "The <content> is not provided which defaults to 'unset'. "
             "It is strongly recommended that content is given explicitly!",
             UserWarning,
         )
-        usecontent = "depth"
+        usecontent = "unset"
```

Comment:
Why the use of "unset" rather than None?

Reply:
In other words, if `content` is None, upload to Sumo will fail on the validation. So in case of allowing `None` we should raise error, not warn, and not allow it. That would be new behavior and possibly breaking change for many. So that must be a bit better planned if we want to do that, and possibly associated with a deprecation warning for a while.